config/macos_clang_x86_64.mk

```diff
@@ -0,0 +1,25 @@
+BUILDDIR:=macos/clang/x86_64
+
+include config/base.mk
+include config/with-clang.mk
+include config/with-debug.mk
+include config/with-brutality.mk
+include config/with-optimization.mk
+include config/with-threads.mk
+
+# Clang sadly doesn't support important optimizations.  This practically
+# limits clang usage to code hygenine usage for the time being.  Here,
+# ideally would do:
+#
+# -falign-functions=32 -falign-jumps=32 -falign-labels=32 -falign-loops=32
+# -mbranch-cost=5
+
+CPPFLAGS+=-fomit-frame-pointer -march=haswell -mtune=skylake -mfpmath=sse \
+	  -DFD_HAS_INT128=1 -DFD_HAS_DOUBLE=1 -DFD_HAS_ALLOCA=1 -DFD_HAS_X86=1 -DFD_HAS_SSE=1 -DFD_HAS_AVX=1
```

Comment:
Not sure but, if it is an x86 CPU MAC, I'd be mildly surprised if AVX support wasn't there.

That is, I'd guess the CPU itself does have it physically in that the level of AVX-ness required for Firedancer as is today was widespread by the mid 2010s (though having a custom SKU made for large enough volume customer isn't completely unheard of).

Likewise, I'd guess the OS does have it logically because even PowerPC CPU used in the pre-x86 CPU MAC era had features like Altivec for vectorization (though it is possible that the OS itself doesn't allow user space applications to play with AVX features for power or context switching performance or vendor flexibility or some other legacy compat reasons).

Reply:
'Recent' x64/x86 Macs have AVX support, but ARM Macs do not have AVX support, so I've omitted AVX flags in the mac ARM Makefile, but included them in the x64 one.

I could find out when Mac (x64) added AVX support and document that. I figured that in the x64 case, use of 'older' Macs without AVX would be pretty unlikely.